.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+
+The [documentation](../documentation) directory in this repo covers the more granular technical aspects of this project. Of particular note for new folks are the following:
+
+- [Guides](../documentation/guides): a set of guides to help you get started developing with `web-foundation`. Of particular note for developers just starting on the project is our guide to [creating a new package](../documentation/guides/creating-a-new-package.md).
+- [FAQ](../documentation/FAQ.md): common questions about the project in general, as well as some of the technical pieces within.
+- [Resources](../documentation/resources.md): good resources for understanding this project’s tech stack.
+- [Getting started](../documentation/getting-started.md): some tools we recommend for getting the most out of this project.
+
+## Testing your changes in a local project
+
+To try out your changes in another locally cloned project, you can use `yarn tophat <package-name-without-@shopify-prefix> <relative-path-to-project>`. Using this command rather than `yarn link` will set up a watcher let you make changes without needing to rerun any commands.
+
+Example: To test my changes to `@shopify/react-form-state` in my local project named `cool-proj`, I would run `yarn tophat react-form-state ../path/to/cool-proj`.
+
+More usage instructions on the `tophat` command can be [found here](https://github.com/Shopify/webgen/blob/master/docs/TOPHAT.md).
+
+### Emoji commits
+
+We have found that prefacing a commit message or PR title with an emoji can be a great way to improve the developer experience when browsing the repo code. Additionally, it is a terse way to convey information. Many of our contributors have found the guide at https://gitmoji.carloscuesta.me/ to be helpful in preserving this dynamic.
+
+### Documentation
+
+If your change affects the public API of any packages within Web-Foundation (i.e. adding or
+changing arguments to a function, adding a new function, changing the
+return value, etc), please ensure the documentation is also updated to
+reflect this. Documentation is in the `README.md` files of each package. If further documentation is needed please communicate via Github Issues.
+
+### Testing
+
+The packages in Web-Foundation are used in mission-critical production scenarios. As such, we try not to merge any untested code. The coverage doesn't strictly need to be 100% across the board, but testing should remain a primary concern.
+
+To run the full test suite, simply run `dev test`. In order for tests to run properly, you may need to first run `dev build`
```

Comment:
`dev test` and `dev build` commands will need to exist.

Reply:
`dev build` doesn't exist. We don't have a typescript build for any of these packages.